java/compaction/compaction-job-creation/src/test/java/sleeper/compaction/strategy/impl/SizeRatioCompactionStrategyTest.java

```diff
@@ -195,49 +146,30 @@ public void shouldCreateJobWithLessThanBatchSizeNumberOfFiles() {
-            FileInfo fileInfo = FileInfo.wholeFile()
-                    .filename("file-" + i)
-                    .partitionId(partition.getId())
-                    .numberOfRecords((long) sizes.get(i))
-                    .build();
+            FileInfo fileInfo = factory.rootFile("file-" + i, (long) sizes.get(i));
             fileInfos.add(fileInfo);
         }
         List<FileInfo> shuffledFileInfos = new ArrayList<>(fileInfos);
         Collections.shuffle(shuffledFileInfos);
 
         // When
-        List<CompactionJob> compactionJobs = sizeRatioCompactionStrategy.createCompactionJobs(Collections.emptyList(), shuffledFileInfos, partitions);
+        List<CompactionJob> compactionJobs = strategy.createCompactionJobs(List.of(), shuffledFileInfos, partitions.buildList());
 
         // Then
         assertThat(compactionJobs).hasSize(3);
-
-        List<String> filesForJob1 = new ArrayList<>();
-        for (int i = 0; i < 5; i++) {
-            filesForJob1.add(fileInfos.get(i).getFilename());
-        }
-        checkJob(compactionJobs.get(0), filesForJob1, partition.getId(), instanceProperties.get(FILE_SYSTEM));
-
-        List<String> filesForJob2 = new ArrayList<>();
-        for (int i = 5; i < 10; i++) {
-            filesForJob2.add(fileInfos.get(i).getFilename());
-        }
-        checkJob(compactionJobs.get(1), filesForJob2, partition.getId(), instanceProperties.get(FILE_SYSTEM));
-
-        List<String> filesForJob3 = new ArrayList<>();
-        for (int i = 10; i < 13; i++) {
-            filesForJob3.add(fileInfos.get(i).getFilename());
-        }
-        checkJob(compactionJobs.get(2), filesForJob3, partition.getId(), instanceProperties.get(FILE_SYSTEM));
+        checkJob(compactionJobs.get(0), fileInfos.subList(0, 5));
+        checkJob(compactionJobs.get(1), fileInfos.subList(5, 10));
+        checkJob(compactionJobs.get(2), fileInfos.subList(10, 13));
     }
 
-    private void checkJob(CompactionJob job, List<String> files, String partitionId, String fileSystem) {
+    private void checkJob(CompactionJob job, List<FileInfo> files) {
         CompactionJob expectedCompactionJob = CompactionJob.builder()
                 .tableId("table-id")
                 .jobId(job.getId()) // Job id is a UUID so we don't know what it will be
-                .partitionId(partitionId)
-                .inputFiles(new ArrayList<>(files))
+                .partitionId("root")
+                .inputFiles(files.stream().map(FileInfo::getFilename).collect(Collectors.toList()))
                 .isSplittingJob(false)
-                .outputFile(fileSystem + "databucket/table-id/partition_" + partitionId + "/" + job.getId() + ".parquet")
+                .outputFile("file://databucket/table-id/partition_root/" + job.getId() + ".parquet")
                 .iteratorClassName(null)
                 .iteratorConfig(null).build();
         job.getInputFiles().sort(Comparator.naturalOrder());
```

Comment:
This could be done by including a call to Stream.sorted when setting the list in the builder.

Reply:
This is really awkward. If we use isEqual then we can be sure we're asserting on everything, but if we make split assertions we can ignore the order.

I prefer isEqual in general. I think I'd rather assert on the order, because there is a particular order in which the files are added to the job, and that is significant to the behaviour.

Reply:
There's an extra difficulty in that the order that the files are added to the job is not stable, since there are several files with the same precedence and different names, and the files are shuffled in the test before it starts. I'd remove the shuffle call from the tests as it adds an element of randomness and therefore flakiness to the tests.

Reply:
Sure, i'll remove the shuffling from the tests and keep the equals assertion